tpot/base.py

```diff
@@ -391,6 +396,55 @@ def _setup_config(self, config_dict):
         else:
             self._config_dict = self.default_config_dict
 
+    def _setup_preprocess_config(self, config_dict):
+        if self.template is None:
+            self.template = 'RandomTree'
+        elif 'PreprocessTransformer' in self.template:
+            return
+        if config_dict:
+            # check for valid keys...
+            preprocess_keys = ['text_columns', 'numeric_columns', 'categorical_columns', 'impute']
+            column_transform_keys = ['text_columns', 'numeric_columns', 'categorical_columns', 'text_transformer', 'categorical_transformer']
+            check_keys_ = []
+            column_transform_ = []
+            for k in config_dict.keys():
+                if k not in preprocess_keys:
+                    check_keys_.append(k)
+                if k in column_transform_keys:
+                    column_transform_.append(k)
+
+            if len(check_keys_) > 0:
+                raise Exception(
+                    'It appears preprocessing configuration contains invalid keys: {}.'
+                    'Valid keys are text_columns, numeric_columns,'
+                    'categorical_columns, impute.'.format(', '.join(check_keys_))
+                )
+
+            # override some settings...
+            if config_dict.get('impute', None) is False:
+                self._fitted_imputer = IdentityTransformer()
+            
+            if len(column_transform_) > 0:
+                column_transform_dict = {}
+                for k in ['text_columns', 'numeric_columns', 'categorical_columns']:
+                    if config_dict.get(k) is not None:
+                        column_transform_dict[k] = [config_dict[k]]
+                for k in ['text_transformer', 'categorical_transformer']:
+                    if k in config_dict:
+                        # force transformers to be a list
+                        if config_dict[k] is list:
+                            column_transform_dict[k] = config_dict[k]
+                        else:
+                            column_transform_dict[k] = [config_dict[k]]
+            self._config_dict['tpot.builtins.PreprocessTransformer'] = column_transform_dict
```

Comment:
This injection could be dangerous - do we have opinions on how it is supposed to be handled?

Reply:
I think `preprocess_config_dict` should be a argument within PreprocessTransformer instead of TPOT. And users should be able to customize it via `config_dict`.

Reply:
Yes this is certainly possible (and possible right now with no changes to TPOT master technically) via the use of templates - I think the question arises related to #507; if its possible to have a "built-in" configuration with text or not.

Maybe the answer is we can't